activerecord/test/cases/signed_id_test.rb

```diff
@@ -223,6 +227,54 @@ def set_signed_id
     Account.signed_id_verifier = old_verifier
   end
 
+  test "Account uses legacy signed ID verifier when secret is set for the model" do
+    account_signed_id_with_message_verifiers = @account.signed_id
+    assert_equal @account, Account.find_signed(account_signed_id_with_message_verifiers)
+
+    ActiveRecord.deprecator.silence do
+      Account.signed_id_verifier_secret = "account secret"
+    end
+    account_signed_id_with_secret = @account.signed_id
+    assert_equal @account, Account.find_signed(account_signed_id_with_secret)
+
+    assert_not_equal account_signed_id_with_message_verifiers, account_signed_id_with_secret
+  ensure
+    ActiveRecord.deprecator.silence do
+      Account.signed_id_verifier_secret = nil
```

Comment:
@jonathanhefner Since `signed_id_verifier_secret uses` a `class_attribute`, even reassigning it to `nil` cannot completely clean up the side effects. For example:
```ruby
Account.signed_id_verifier_secret = "account secret"
Account.signed_id_verifier_secret = nil
ActiveRecord::Base.signed_id_verifier_secret = "secret"

puts Account.signed_id_verifier_secret # => nil
```
Do you know any solution or we can ignore it for now?